msg/templates/urtps/microRTPS_timesync.h.em

```diff
@@ -211,4 +211,34 @@ private:
 	 * @@param[in] offset The value of the offset to update to
 	 */
 	inline void updateOffset(const uint64_t& offset) { _offset_ns.store(offset, std::memory_order_relaxed); }
+
+	/** Timesync msg Getters **/
+@[if fastrtps_version <= 1.7 or not ros2_distro]@
+	inline uint64_t getMsgTimestamp(const timesync_msg_t* msg) { return msg->timestamp_(); }
+	inline uint8_t getMsgSysID(const timesync_msg_t* msg) { return msg->sys_id_(); }
+	inline uint8_t getMsgSeq(const timesync_msg_t* msg) { return msg->seq_(); }
+	inline int64_t getMsgTC1(const timesync_msg_t* msg) { return msg->tc1_(); }
+	inline int64_t getMsgTS1(const timesync_msg_t* msg) { return msg->ts1_(); }
+@[elif ros2_distro]@
+	inline uint64_t getMsgTimestamp(const timesync_msg_t* msg) { return msg->timestamp(); }
+	inline uint8_t getMsgSysID(const timesync_msg_t* msg) { return msg->sys_id(); }
+	inline uint8_t getMsgSeq(const timesync_msg_t* msg) { return msg->seq(); }
+	inline int64_t getMsgTC1(const timesync_msg_t* msg) { return msg->tc1(); }
+	inline int64_t getMsgTS1(const timesync_msg_t* msg) { return msg->ts1(); }
+@[end if]@
+
+	/** Timesync msg Setters **/
+@[if fastrtps_version <= 1.7 or not ros2_distro]@
+	inline uint64_t setMsgTimestamp(timesync_msg_t* msg, const uint64_t& timestamp) { msg->timestamp_() = timestamp; }
```

Comment:
For the timestamp, sysID and Seq, can't you just use the generic versions you defined up above? I think you only need specific overrides here for the TC1 and TS1 fields

Reply:
@jkflying `RtpsTopics` is not a level above the `microRTPS_timesync`. Actually ``RtpsTopics.h` includes `microRTPS_timesync.h`. My idea overall is to define an header metadata msg that can be included in all messages, and then I could generalise the setters and getters for it in an upper level header file.